File: twister2/python-support/src/main/python/twister2/tset/fn/factory/TSetFunctions.py
from twister2.tset.fn.factory.GenericFunctions import GenericFunctions
from twister2.tset.fn.factory.PartitionFunctions import PartitionFunctions


class TSetFunctions:

    def __init__(self, java_ref, env):
        self.__java_ref = java_ref
        self.__partition_functions = PartitionFunctions(java_ref.partition(), env)
        self.__map_functions = GenericFunctions(java_ref.map(), env)
        self.__compute_functions = GenericFunctions(java_ref.compute(), env)
        self.__compute_collector_functions = GenericFunctions(java_ref.computeCollector(), env)
        self.__sink_functions = GenericFunctions(java_ref.sink(), env)
        self.__reduce_functions = GenericFunctions(java_ref.reduce(), env)
        self.__apply_functions = GenericFunctions(java_ref.apply(), env)
        self.__flat_map_functions = GenericFunctions(java_ref.flatMap(), env)
        self.__comparator_functions = GenericFunctions(java_ref.comparator(), env)

    @property
    def partition(self) -> PartitionFunctions:
        return self.__partition_functions

    @property
    def map(self) -> GenericFunctions:
        return self.__map_functions

    @property
    def flat_map(self) -> GenericFunctions:
        return self.__flat_map_functions

    @property
    def compute(self) -> GenericFunctions:
        return self.__compute_functions

    @property
    def compute_with_collector(self) -> GenericFunctions:
        return self.__compute_collector_functions

    @property
    def sink(self) -> GenericFunctions:
        return self.__sink_functions

    @property
    def apply(self) -> GenericFunctions:
        return self.__apply_functions

    @property
    def reduce(self) -> GenericFunctions:
        return self.__reduce_functions

    @property
    def comparator(self) -> GenericFunctions:
        return self.__comparator_functions

File: twister2/python-support/src/main/java/edu/iu/dsc/tws/python/tset/fn/TSetFunctions.java
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
package edu.iu.dsc.tws.python.tset.fn;

public final class TSetFunctions {

  private static final TSetFunctions INSTANCE = new TSetFunctions();

  private TSetFunctions() {

  }

  public static TSetFunctions getInstance() {
    return INSTANCE;
  }

  public PartitionFunctions partition() {
    return PartitionFunctions.getInstance();
  }

  public MapFunctions map() {
    return MapFunctions.getInstance();
  }

  public FlatMapFunctions flatMap() {
    return FlatMapFunctions.getInstance();
  }

  public ComputeFunctions compute() {
    return ComputeFunctions.getInstance();
  }

  public ComputeWithCollectorFunctions computeCollector() {
    return ComputeWithCollectorFunctions.getInstance();
  }

  public SinkFunctions sink() {
    return SinkFunctions.getInstance();
  }

  public ReduceFunctions reduce() {
    return ReduceFunctions.getInstance();
  }

  public ApplyFunctions apply() {
    return ApplyFunctions.getInstance();
  }
}
